panther_analysis_tool/command/migrate.py

```diff
@@ -50,53 +55,161 @@ class MigrationItem:
     "Items deleted"
 
     def empty(self) -> bool:
-        return len(self.items_with_conflicts) == 0 and len(self.items_migrated) == 0
+        return (
+            len(self.items_with_conflicts) == 0
+            and len(self.items_migrated) == 0
+            and len(self.items_deleted) == 0
+        )
 
-    def _by_analysis_type(self, items: list[MigrationItem]) -> dict[str, list[MigrationItem]]:
-        result = defaultdict(list)
-        for item in items:
-            result[item.pretty_analysis_type].append(item)
-        return dict(result)
+    def has_conflicts(self) -> bool:
+        return len(self.items_with_conflicts) > 0
 
-    def migrated_items_by_analysis_type(self) -> dict[str, list[MigrationItem]]:
-        return self._by_analysis_type(self.items_migrated)
+    def items_with_conflicts_sorted(self) -> list[MigrationItem]:
+        return sorted(
+            self.items_with_conflicts.values(), key=lambda x: x.pretty_analysis_type + x.analysis_id
+        )
 
-    def items_with_conflicts_by_analysis_type(self) -> dict[str, list[MigrationItem]]:
-        return self._by_analysis_type(self.items_with_conflicts)
+    def items_deleted_sorted(self) -> list[MigrationItem]:
+        return sorted(
+            self.items_deleted.values(), key=lambda x: x.pretty_analysis_type + x.analysis_id
+        )
 
-    def has_conflicts(self) -> bool:
-        return len(self.items_with_conflicts) > 0
+    def items_migrated_sorted(self) -> list[MigrationItem]:
+        return sorted(
+            self.items_migrated.values(), key=lambda x: x.pretty_analysis_type + x.analysis_id
+        )
+
+    def migrated_merged_items(self) -> list[analysis_utils.AnalysisItem]:
+        return [
+            item.merged_item
+            for item in self.items_migrated.values()
+            if item.merged_item is not None
+        ]
+
+    def to_dict(self) -> dict:
+        """Returns a JSON string representation of the migration status."""
+        return {
+            "items_with_conflicts": {k: v.to_dict() for k, v in self.items_with_conflicts.items()},
+            "items_migrated": {k: v.to_dict() for k, v in self.items_migrated.items()},
+            "items_deleted": {k: v.to_dict() for k, v in self.items_deleted.items()},
+        }
+
+    @staticmethod
+    def from_dict(dict: dict) -> "MigrationStatus":
+        """Returns a MigrationStatus from a JSON dictionary."""
+        return MigrationStatus(
+            items_with_conflicts={
+                k: MigrationItem.from_dict(v)
+                for k, v in dict.get("items_with_conflicts", {}).items()
+            },
+            items_migrated={
+                k: MigrationItem.from_dict(v) for k, v in dict.get("items_migrated", {}).items()
+            },
+            items_deleted={
+                k: MigrationItem.from_dict(v) for k, v in dict.get("items_deleted", {}).items()
+            },
+        )
+
+    def write_migration_status(self) -> None:
```

Comment:
This is generally the same. Just moved into a method instead of a function. It also adds serializing to json.